libraries/Microsoft.Bot.Builder.Expressions/BuiltInFunctions.cs

```diff
@@ -630,18 +598,44 @@ private static (object value, string error) ExtractElement(Expression expression
                         }
                         else
                         {
-                            error = $"{instance} is not a collection.";
+                            error = $"type of ({instance}, {idx}) should be (object, string) or (collection, int)";
```

Comment:
Actually, the original "{instance} is not a collection." was the right string--sort

Reply:
I made this mistake by missing the scoping in the diff--sorry!